app/src/main/java/com/kamwithk/ankiconnectandroid/ankidroid_api/Utility.java

```diff
@@ -0,0 +1,79 @@
+package com.kamwithk.ankiconnectandroid.ankidroid_api;
+
+import android.text.Html;
+import java.math.BigInteger;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public final class Utility {
+
+    private static Pattern STYLE_PATTERN = Pattern.compile("(?s)<style.*?>.*?</style>");
+    private static Pattern SCRIPT_PATTERN = Pattern.compile("(?s)<script.*?>.*?</script>");
+    private static Pattern TAG_PATTERN = Pattern.compile("<.*?>");
+    private static Pattern IMG_PATTERN = Pattern.compile("<img src=[\"']?([^\"'>]+)[\"']? ?/?>");
+    private static Pattern HTML_ENTITIES_PATTERN = Pattern.compile("&#?\\w+;");
+    private static final String FIELD_SEPARATOR = Character.toString('\u001f');
+
+    private Utility() {
+    }
+
+    // taken from AnkiDroid
+    public static String[] splitTags(String tags) {
+        if (tags == null) {
+            return null;
+        }
+        return tags.trim().split("\\s+");
+    }
+
+    public static String[] splitFields(String fields) {
+        return fields != null? fields.split(FIELD_SEPARATOR, -1): null;
+    }
+
+    public static long getFieldChecksum(String data) {
+        String strippedData = stripHTMLMedia(data);
+
+        try {
+            MessageDigest md = MessageDigest.getInstance("SHA1");
+            byte[] digest = md.digest(strippedData.getBytes(StandardCharsets.UTF_8));
+            BigInteger bigInteger = new BigInteger(1, digest);
+            String result = bigInteger.toString(16);
+
+            if(result.length() < 40) {
+                String zeroes = "0000000000000000000000000000000000000000";
```

Comment:
Why do you have a string with exactly 40 zeros here? I assume this is because sha-1 is a 40 character long hexadecimal number?
This may be better as a constant named SHA1_ZEROS or something along those lines

Reply:
My understanding is - the checksum of a note is the "32 bit unsigned number from first 8 digits of sha1 hash". Anki desktop doesn't pad the string to a length of 40 from what I can see, but AnkiDroid does in its query API (if you add a note in the app itself, I believe it just uses the same function call that Anki desktop uses).

This code was taken from AnkiDroid to make sure the checksum calculation for the notes is the same.

Reply:
Thanks for explaining, makes sense